includes/privacy/class-wcs-privacy.php

```diff
@@ -124,7 +124,7 @@ public static function add_privacy_bulk_action( $bulk_actions ) {
 	 */
 	public static function handle_privacy_bulk_actions( $redirect_url, $action, $subscription_ids ) {
 		if ( 'wcs_remove_personal_data' !== $action ) {
-			return;
+			return $redirect_url;
```

Comment:
This line fixes the main notice issue.

The other changes are a little refactor. As the `handle_privacy_bulk_actions` function is called by the `handle_bulk_actions-woocommerce_page_wc-orders--shop_subscription` hook as well, removed the redirection and exit from this function, moving it to the other parent (except the hook) function `process_bulk_action`.